src/config/mod.rs

```diff
@@ -1326,9 +1327,12 @@ impl DbConfig {
                 self.enable_multi_batch_write.get_or_insert(false);
                 self.allow_concurrent_memtable_write.get_or_insert(false);
                 let total_mem = SysQuota::memory_limit_in_bytes() as f64;
-                self.write_buffer_limit.get_or_insert(ReadableSize(
+                // purge-threshold is set to twice the limit. Too large limit will cause trouble
+                // to raft log replay.
+                self.write_buffer_limit.get_or_insert(ReadableSize(cmp::min(
                     (total_mem * WRITE_BUFFER_MEMORY_LIMIT_RATE) as u64,
-                ));
+                    WRITE_BUFFER_MEMORY_LIMIT_MAX,
+                )));
```

Comment:
remember to add a release note and update docs on the WRITE_BUFFER_MEMORY_LIMIT_MAX limit

Reply:
So the write buffer memory limit can still be larger than 15GB if user sets it explicitly, right?
I think maybe we can just simplify the logic by reduce the WRITE_BUFFER_MEMORY_LIMIT_RATE from 0.25 to 0.2. Then in common spec 64GB it will be 12.8G < 15G. That should cover most of the cases. 
The logic itself in this PR is ok, but it's a bit harder to document it clearly.

Reply:
It's the same logic as rocksdb.max-background-jobs or backup.num-threads. Changing the rate is a bigger change that requires a thorough performance compare.

Reply:
BTW it also can't be documented because it is dependent on `storage.engine`, which is not a public config yet.